### PR TITLE
implement global context

### PR DESCRIPTION
**Pull request by tiancizhong**
For global variable: only assignment is allowed such as `a = 3, b = 4;` and a block of global assignment need to end with double `;` 
example
```
a = 3,
b = 4;
SomeContext = {}
c = 4;
SomeContext2 = {}
```
For each block of global variable an cxt_del named "Global" will be saved. There might be multi cxt_del named "Global" in the `contexts` (the list of context decl). In the semant, all the cxt_del named "Global" will be collapsed into one cxt_del named "Global". We will create a new list of `contexts`  (the list of context decl) that contains only one cxt_del named "Global" instead of multiple cxt_del named "Global". And the content of the "Global" context to all the context.

### CODE REVIEW

**Comment by tiancizhong** [comment]
```
  List.iter check_ctx_break_continue (new_contexts contexts);
  List.iter check_find_break_continue finds;
  List.iter check_range finds;
  List.iter check_ctx contexts;
  List.iter check_find finds;

  {
    Sast.ast = ((new_contexts contexts), finds);
    Sast.vars = varmap;
    Sast.lib = liblist
  }
``` 
This is what I am trying to do and get error message for all our test:
`Fatal error: exception Not_found`
Not sure how to fix the error and could not find where the error come from.

**Comment by jzacsh** [comment]
sorry, i think i need to sit down and look at all the code you're trying to use. would be easier for me if you want to just meet - or even video chat. just ping me in slack when you have time to walk me through what you're stuck on.

**Comment by tiancizhong** [comment]
One of the test-global is skipped because codegen needs a fully update. The variable declaration need to be came from sast instead of varmap. @jzacsh  do you mind to take a look at this pull request to see if it is ready to merge?

**Comment by rxie25** [comment]
looks good.

**Comment by jzacsh** [comment]
will take a look later tonight

File: src/scanner.mll [hunk 1]
```diff
@@ -22,7 +22,7 @@ let newline = '\n' | "\r\n"
 let reserved = ("int" | "double" | "char" | "float" | "const" | "void" | "short" | "struct" 
 				| "long" | "return" | "static" | "swtich" | "case" | "default" | "for" | "do" | "go to"
 				| "auto" | "signed" | "extern" | "register" | "enum" | "sizeof" | "typedef" | "union" 
-				| "volatile")
+				| "volatile" | "Global")
```
Comment (jzacsh) [inline src/scanner.mll]: nice! can you add this to the LRM, in this pull request before merging?

File: src/tests/test-global.skip.eq [hunk 2]
```diff
@@ -0,0 +1,11 @@
+skipb = 23,
```
Comment (jzacsh) [inline src/tests/test-global.skip.eq]: nice! maybe - to clarify the distinction between these two tests - rename this one to something like `test-mix-global-named-ctx` (where as the other one is *only* testing global context behavior, right?)

Comment (jzacsh) [inline src/tests/test-global.skip.eq]: also, can you add (whatever the name of this test ends up being) to the check-list in issue #86 ?

**Comment by jzacsh** [comment]
aside from my comments, looks good to me! feel free to merge, once they're resolved _(or acknowledged and ignored if you want)_!

File: notes/language-reference-manual.md [hunk 3]
```diff
@@ -168,6 +168,35 @@ Following are reserved keywords, and have special meaning in the language. See
 + `else`
 + `find`
 + `print`
++ `Global`
+
+Eqeq is translated into C, so it has the same set of reserved keyword as in C in additional to the reserved keywords above.
```
Comment (jzacsh) [inline notes/language-reference-manual.md]: nice catch!

**Pull request merged**